### PR TITLE
S3: Use key_id and key_secret directly

### PR DESCRIPTION
**Pull request by farizrahman4u**
Fixes https://github.com/iterative/dvc/issues/4175

@efiop

### CODE REVIEW

**Comment by efiop** [comment]
Hi @farizrahman4u !

Thank you for the PR! Please take a look at our contribution guide (it was in the PR hints). Right now our patch checks are failing, because you didn't have pre-commit hooks installed. :slightly_smiling_face:

File: dvc/tree/s3.py [hunk 1]
```diff
@@ -54,18 +54,32 @@ def __init__(self, repo, config):
...
         shared_creds = config.get("credentialpath")
         if shared_creds:
             os.environ.setdefault("AWS_SHARED_CREDENTIALS_FILE", shared_creds)
+        self._validate_config()
+
+    def _validate_config(self):
+        if bool(self.key_id) != bool(self.key_secret):
+            raise DvcException(
+                "Provide either both or none of `s3_key_id` and `s3_key_secret`."
+            )
```
Comment (efiop) [inline dvc/tree/s3.py:69]: Does `boto3.session` require you pass both at the same time? Or will find the other from env/configs if you only give it one of the keys?

File: dvc/tree/s3.py [hunk 2]
```diff
@@ -54,18 +54,32 @@ def __init__(self, repo, config):
...
-        session = boto3.session.Session(
-            profile_name=self.profile, region_name=self.region
-        )
+        session_opts = dict(profile_name=self.profile, region_name=self.region)
+
+        if self.key_id and self.key_secret:
+            session_opts["aws_access_key_id"] = self.key_id
+            session_opts["aws_secret_access_key"] = self.key_secret
+
+        session = boto3.session.Session(**session_opts)
```
Comment (efiop) [inline dvc/tree/s3.py:82]: Let's add a simple unit test to`tests/unit/remote/test_s3.py`

File: dvc/tree/s3.py [hunk 3]
```diff
@@ -54,18 +54,32 @@ def __init__(self, repo, config):
 
         self._append_aws_grants_to_extra_args(config)
 
+        self.key_id = config.get("s3_key_id")
+        self.key_secret = config.get("s3_key_secret")
+
```
Comment (efiop) [inline dvc/tree/s3.py:59]: I don't think this is going to work, as you didn't enable these fields in `dvc/config.py`.

**Comment by efiop** [comment]
Also, please feel free to mark your PR as WIP/draft if it is not ready for review yet :slightly_smiling_face: Thanks!

**Comment by farizrahman4u** [comment]
@efiop CI seems to be cloning the main repo instead of the fork?

**Comment by skshetry** [comment]
Looks like it is trying to run restyled's PR and is failing as the branch is no longer there. 
@farizrahman4u, please rebase this PR. Probably, that will fix the issue.

**Comment by farizrahman4u** [comment]
@efiop Done.

File: dvc/config.py [hunk 4]
```diff
@@ -149,6 +149,8 @@ class RelPath(str):
                     "profile": str,
                     "credentialpath": str,
                     "endpointurl": str,
+                    Optional("s3_key_id", default=""): str,
+                    Optional("s3_key_secret", default=""): str,
```
Comment (efiop) [inline dvc/config.py:153]: These are optional by default, you don't need to set them
```suggestion
                    "s3_key_id": str,
                    "s3_key_secret": str,
```

Comment (farizrahman4u) [reply dvc/config.py]: done

File: tests/unit/remote/test_s3.py [hunk 5]
```diff
@@ -6,6 +6,8 @@
 bucket_name = "bucket-name"
 prefix = "some/prefix"
 url = f"s3://{bucket_name}/{prefix}"
+key_id = "key_id"
+key_secret = "key_secret"
```
Comment (efiop) [inline tests/unit/remote/test_s3.py:10]: a bit worried of some nasty bug like that :smiley: 
```suggestion
key_id = "key-id"
key_secret = "key-secret"
```

File: dvc/config.py [hunk 6]
```diff
@@ -149,6 +149,8 @@ class RelPath(str):
                     "profile": str,
                     "credentialpath": str,
                     "endpointurl": str,
+                    "s3_key_id": str,
+                    "s3_key_secret": str,
```
Comment (efiop) [inline dvc/config.py:153]: Kinda have an urge to just do
```suggestion
                    "endpointurl": str,
                    "key_id": str,
                    "key_secret": str,
```
or, even better:
```suggestion
                    "endpointurl": str,
                    "access_key_id": str,
                    "secret_access_key": str,
```
to make it easier to find and more intuitive in general. Considering that `oss` has
```
                    "oss_key_id": str,
                    "oss_key_secret": str,
                    "oss_endpoint": str,
```
already (which we could consider unifying in the future).

**Comment by efiop** [comment]
Btw, @farizrahman4u , in the PR header that you've removed, there was a bulletpoint about docs. In particular, we need to add the new config options to https://dvc.org/doc/command-reference/remote/modify (there is an edit button). Could you please submit a docs PR after we are done with config option naming here? :slightly_smiling_face:

**Comment by farizrahman4u** [comment]
@efiop done renames and docs pr : https://github.com/iterative/dvc.org/pull/1589

File: dvc/tree/s3.py [hunk 7]
```diff
@@ -54,6 +54,9 @@ def __init__(self, repo, config):
 
         self._append_aws_grants_to_extra_args(config)
 
+        self.access_key_id = config.get("access_key_id")
+        self.secret_access_key = config.get("secret_access_key")
```
Comment (Suor) [inline dvc/tree/s3.py:58]: Why do we assign it instead of using `self.config` directly later?

Comment (farizrahman4u) [reply dvc/tree/s3.py:58]: Following the pattern in other remotes (OSS, GDrive etc).

**Pull request merged**